common/telemetry/grpc.go

```diff
@@ -52,32 +55,56 @@ type (
 func NewServerTraceInterceptor(
 	tp trace.TracerProvider,
 	tmp propagation.TextMapPropagator,
+	logger log.Logger,
 ) ServerTraceInterceptor {
 	//nolint:staticcheck
 	otelInterceptor := otelgrpc.UnaryServerInterceptor(
 		otelgrpc.WithPropagators(tmp),
 		otelgrpc.WithTracerProvider(tp),
 	)
+	return annotationInterceptor(otelInterceptor, logger)
+}
 
-	if debugMode() {
-		return func(
-			ctx context.Context,
-			req interface{},
-			info *grpc.UnaryServerInfo,
-			handler grpc.UnaryHandler,
-		) (any, error) {
-			return otelInterceptor(ctx, req, info, debugHandler(handler))
-		}
-	}
+func annotationInterceptor(
+	otelInterceptor grpc.UnaryServerInterceptor,
+	logger log.Logger,
+) ServerTraceInterceptor {
+	return func(
+		ctx context.Context,
+		req interface{},
+		info *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (any, error) {
+		isDebug := debugMode()
+		tags := logtags.NewWorkflowTags(common.NewProtoTaskTokenSerializer(), logger)
```

Comment:
This is re-using the existing generated code that can extract Workflow tags from gRPC requests. It seems to do exactly what we need, so I'm not inclined to clone/fork that code.